prov/shm/src/smr_progress.c

```diff
@@ -1192,7 +1193,7 @@ static void smr_progress_ipc_list(struct smr_ep *ep)
 		}
 
 		ret = smr_complete_rx(ep, context, ipc_entry->cmd.msg.hdr.op,
-				flags, ipc_entry->bytes_done,
+				flags, bytes_done,
```

Comment:
I think the issue in the old code is that we report the size of the completion as being ipc_entry->bytes_done but it is never actually getting set. 
I think this is doing it in a roundabout way though. The rocr code here isn't tracking how many bytes were actually copied but it's just setting an internal value to be the size that was submitted and then reporting it back. I think it makes more sense for shm to be able to track that on its own since it knows what transfer that corresponds to. The total size of the message (that was submitted for copying) should be in the cmd->msg.hdr.size which you have available to you at the time of the completion reporting.
Could this issue be fixed by just changing this one parameter here to `ipc_entry->cmd.msg.hdr.size`?

Reply:
Actually, I think you're right. we can just report that since when the async operations complete we should've completed the entire copy.